lib/util/fs/index.ts

```diff
@@ -11,6 +11,18 @@ export * from './proxies';
 
 export const pipeline = util.promisify(stream.pipeline);
 
+function isPathInBaseDir(path: string, baseDir: string): boolean {
+  if (!path.startsWith(upath.resolve(baseDir))) {
+    logger.warn(
+      { path, baseDir },
+      'Preventing access to file outside the base directory'
+    );
+    return false;
+  }
+
+  return true;
```

Comment:
@rarkins @zharinov we should probably throw an error here and abort further processing.

so function shou be rename to `validatePathInBaseDir` or something like that and return never?

Reply:
"abort further processing" = stop running on the repo?

Reply:
yes

Reply:
Sounds reasonable + further we need to ensure we don't use `fs` and `fs-extra` directly (i.e. migrate and add lint rule)